pymeasure/instruments/rohdeschwarz/hmp.py

```diff
@@ -0,0 +1,271 @@
+#
+# This file is part of the PyMeasure package.
+#
+# Copyright (c) 2013-2022 PyMeasure Developers
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+import logging
+from typing import List
+
+from pymeasure.instruments import Instrument
+from pymeasure.instruments.validators import (strict_discrete_set,
+                                              truncated_range)
+
+log = logging.getLogger(__name__)
+log.addHandler(logging.NullHandler())
+
+
+def process_sequence(sequence: List[float]) -> str:
+    """
+    Check and prepare sequence data.
+
+    The form of the sequence data is "Voltage1,Current1,Time1,Voltage2,
+    Current2,Time2,...,Voltage128,Current128,Time128" with voltages in V,
+    currents in A, and times in s. Dwell times are between 0.06 and 10 s.
+    """
+    if not len(sequence) % 3 == 0:
+        raise ValueError("Sequence must contain multiple of 3 values.")
+    if any(t > 10 or t < 0.06 for t in sequence[2::3]):
+        raise ValueError("Dwell times must be between 0.06 and 10 s.")
+    # turn sequence data into a string
+    sequence = ",".join(str(s) for s in sequence)
+    return sequence
+
+
+class HMP4040(Instrument):
+    """Represents a Rohde&Schwarz HMP4040 power supply."""
+
+    def __init__(self, resourceName, name="Rohde&Schwarz HMP4040", **kwargs):
+        super().__init__(
+            resourceName, name, includeSCPI=True, **kwargs
```

Comment:
```suggestion
    def __init__(self, adapter, **kwargs):
        kwargs.setdefault("name", "Rohde&Schwarz HMP4040")
        super().__init__(
            adapter, name, includeSCPI=True, **kwargs
```
anticipating some changes already in the pipeline :-)